website/content/docs/plugins/plugin-architecture.mdx

```diff
@@ -0,0 +1,190 @@
+it instead becomes visible to backends and can be executed by them. For more
+information on the plugin catalog please see the [Plugin Catalog API
+docs](/api-docs/system/plugins-catalog).
+
+An example of plugin registration in current versions of Vault:
+
+```shell-session
+$ vault plugin register -sha256=<SHA256 Hex value of the plugin binary> \
+    secret \                  # type
+    myplugin-database-plugin
+
+Success! Registered plugin: myplugin-database-plugin
+```
+
+Vault versions prior to v0.10.4 lacked the `vault plugin` operator and the
+registration step for them is:
+
+```shell-session
+$ vault write sys/plugins/catalog/database/myplugin-database-plugin \
+     sha256=<SHA256 Hex value of the plugin binary> \
+     command="myplugin"
+
+Success! Data written to: sys/plugins/catalog/database/myplugin-database-plugin
+```
+
+### Plugin Execution
+
+When a backend wants to run a plugin, it first looks up the plugin, by name, in
+the catalog. It then checks the executable's SHA256 sum against the one
+configured in the plugin catalog. Finally Vault runs the command configured in
+the catalog, sending along the JWT formatted response wrapping token and mlock
+settings. Like Vault, plugins support [the use of mlock when available](/docs/configuration#disable_mlock).
+
+~> Note: If Vault is configured with `mlock` enabled, then the Vault executable
+and each plugin executable in your [plugins directory](/docs/plugins/plugin-architecture#plugin-directory)
+must be given the ability to use the `mlock` syscall.
+
+### Plugin Upgrades
+
+External plugins may be updated by registering and reloading them. More details
+on the upgrade procedure can be found in
+[Upgrading Vault Plugins](/docs/upgrading/plugins).
+
+## Plugin Multiplexing
+
+Database plugins can be made to implement plugin multiplexing
+which allows a single plugin process to be used for multiple database
+connections. This single process, per database plugin, will be multiplexed
+across all Vault namespaces for mounts of this type. Multiplexing a plugin
+does not affect the current behavior of existing plugins.
+
+To enable multiplexing, the plugin must be compiled with the `ServeMultiplex`
+function call from Vault's `dbplugin` package. At this time, there is no
+opt-out capability for plugins that implement multiplexing. To use a
+non-multiplexed plugin, run an older version of the plugin. i.e. the
+plugin calls the `dbplugin.Serve` function. More details
+on implementing plugin multiplexing can be found in
+[Upgrading Vault Plugins](/docs/secrets/databases/custom#serving-a-plugin-with-multiplexing).
+
+## Troubleshooting
+
+### Unrecognized remote plugin message
+
+If the following error is encountered when enabling a plugin secret engine or
+auth method:
+
+<CodeBlockConfig hideClipboard>
+
+```sh
+Unrecognized remote plugin message:
+
+This usually means that the plugin is either invalid or simply
```

Comment:
```suggestion
This usually means that the plugin is either invalid or 
```

Reply:
I am going to leave this as-is since it is part of the error message returned in this case